github/commands/pull_request.py

```diff
@@ -74,59 +76,113 @@ def on_select_action(self, idx):
         if idx == -1:
             return
 
+        owner = self.pr["head"]["repo"]["owner"]["login"]
+
+        if owner == self.base_remote.owner:
+            # don't add prefix for integrated remote
+            branch_name = self.pr["head"]["ref"]
+        else:
+            branch_name = "{}-{}".format(owner, self.pr["head"]["ref"])
+
         if idx == 0:
             self.fetch_and_checkout_pr()
         elif idx == 1:
             show_single_line_input_panel(
                 "Enter branch name for PR {}:".format(self.pr["number"]),
-                "{}/{}".format(self.pr["user"]["login"], self.pr["head"]["ref"]),
+                branch_name,
                 self.fetch_and_checkout_pr
             )
         elif idx == 2:
             show_single_line_input_panel(
                 "Enter branch name for PR {}:".format(self.pr["number"]),
-                "{}/{}".format(self.pr["user"]["login"], self.pr["head"]["ref"]),
-                self.create_branch_for_pr
+                branch_name,
+                lambda x: self.create_branch_for_pr(x, ask_set_upstream=True)
             )
         elif idx == 3:
             self.view_diff_for_pr()
         elif idx == 4:
             self.open_pr_in_browser()
 
     def fetch_and_checkout_pr(self, branch_name=None):
-        self.window.status_message("Fetching PR commit...")
-        self.git(
-            "fetch",
-            self.pr["head"]["repo"]["clone_url"],
-            self.pr["head"]["ref"]
-        )
-
         if branch_name:
-            self.window.status_message("Creating local branch for PR...")
-            self.git(
-                "branch",
-                branch_name,
-                self.pr["head"]["sha"]
-            )
+            self.create_branch_for_pr(branch_name, checkout=True, ask_set_upstream=True)
+        else:  # detached HEAD
+            self.checkout_detached()
```

Comment:
Remove method; it also has a bug (or is it a feature?) that if the user did not enter a branch name we continue with `checkout_detached` instead of aborting. If it's a feature we should have it of course for case 1 *and* 2 t.i. on L132.